modules/channel-teams/src/backend/client.ts

```diff
@@ -0,0 +1,181 @@
+import { BotFrameworkAdapter, ConversationReference, CardFactory, TurnContext, AttachmentLayoutTypes } from 'botbuilder'
+
+import * as sdk from 'botpress/sdk'
+import { Router } from 'express'
+import _ from 'lodash'
+
+import { Config } from '../config'
+
+import { Clients } from './typings'
+
+const outgoingTypes = ['message', 'typing', 'carousel', 'text']
+
+export class TeamsClient {
+  private conversationsRefs: _.Dictionary<Partial<ConversationReference>> = {}
+  private adapter: BotFrameworkAdapter
+  constructor(private bp: typeof sdk, private botId: string, private config: Config, private router: Router) {}
+
+  async initialize() {
+    this.adapter = new BotFrameworkAdapter({
+      appId: this.config.microsoftAppId,
+      appPassword: this.config.microsoftAppPassword
+    })
+
+    this.router.post('/api/messages', async (req, res) => {
```

Comment:
Is this route secured proprely? I suppose that processActivity checks for auth token and validates the request?

Reply:
Yeah, processActivity will throw unauthorization exception if the request does not contain appId and secret.